qupy/hamiltonian.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+from qupy.operator import X, Y, Z
+
+
+class Hamiltonian:
+    """
+    Creats Hamiltonian as a sum of pauli terms.
+    $ H = \sum_j coefs[j]*ops[j] $
+    Args:
+        n_qubit (:class:`int`):
+            Number of qubits.
+        coefs (:class:`list`):
+            list of floats. coefficient of each pauli terms.
+        ops (:class:`list`):
+            pauli terms
+            each element of this argument is a str specifying the pauli term.
+            e.g. ops[0] = "IIZZI"
+
+    Attributes:
+        n_qubit (:class:`int`):
+            Number of qubits.
+        coefs (:class:`list`):
+            list of floats. coefficient of each pauli terms.
+        ops (:class:`list`):
+            pauli terms
+            each element of this argument is a str specifying the pauli term.
+            e.g. ops[0] = "IIZZI"
+    """
+
+    def __init__(self, n_qubit, coefs=None, ops=None):
+        self.n_qubit = n_qubit
+        self.coefs = coefs if coefs is not None else np.array([])

```

Comment:
According to docstring, self.coefs is list, so please write
`self.coefs = coefs if coefs is not None else []`